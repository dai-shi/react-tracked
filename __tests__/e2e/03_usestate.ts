/* global page */

jest.setTimeout(15 * 1000);

describe('03_usestate', () => {
  const port = process.env.PORT || '8080';

  it('should work with recorded events', async () => {
    await page.goto(`http://localhost:${port}/`);

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button');
    await page.click('body > #app > div:nth-child(2) > div > button');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(2) > div > button');
    await page.click('body > #app > div:nth-child(2) > div > button');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button');
    await page.click('body > #app > div:nth-child(3) > div > button');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button');
    await page.click('body > #app > div:nth-child(3) > div > button');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(3) > div > button');
    await page.click('body > #app > div:nth-child(3) > div > button');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.waitForSelector('body > #app > div:nth-child(5) > div:nth-child(1) > input');
    await page.click('body > #app > div:nth-child(5) > div:nth-child(1) > input');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > div:nth-child(5) > div:nth-child(1) > input', 'a');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > div:nth-child(5) > div:nth-child(1) > input', 'a');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > div:nth-child(5) > div:nth-child(1) > input', 'a');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > div:nth-child(5) > div:nth-child(2) > input', 'b');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > div:nth-child(5) > div:nth-child(2) > input', 'b');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > div:nth-child(5) > div:nth-child(2) > input', 'b');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > div:nth-child(5) > div:nth-child(3) > input', '1');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > div:nth-child(5) > div:nth-child(3) > input', '1');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.type('body > #app > div:nth-child(5) > div:nth-child(3) > input', '1');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();
  });
});
