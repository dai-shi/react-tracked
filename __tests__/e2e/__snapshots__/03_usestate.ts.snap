// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`03_usestate should work with recorded events 1`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 6<div><span>Count: 1</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 8<div><span>Count: 1</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 4<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 2`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 10<div><span>Count: 2</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 12<div><span>Count: 2</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 4<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 3`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 14<div><span>Count: 3</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 16<div><span>Count: 3</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 4<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 4`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 18<div><span>Count: 4</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 20<div><span>Count: 4</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 4<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 5`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 4<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 6`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 2<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 4<div>First Name:<input value=\\"\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 7`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 6<div>First Name:<input value=\\"a\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 8<div>First Name:<input value=\\"a\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 8`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 14<div>First Name:<input value=\\"aa\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 16<div>First Name:<input value=\\"aa\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 9`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 18<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 20<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 10`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 22<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"b\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 24<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"b\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 11`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 26<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bb\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 28<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bb\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 12`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 30<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bbb\\"></div><div>Age:<input value=\\"0\\"></div></div><div>numRendered: 32<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bbb\\"></div><div>Age:<input value=\\"0\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 13`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 34<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bbb\\"></div><div>Age:<input value=\\"1\\"></div></div><div>numRendered: 36<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bbb\\"></div><div>Age:<input value=\\"1\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 14`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 38<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bbb\\"></div><div>Age:<input value=\\"11\\"></div></div><div>numRendered: 40<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bbb\\"></div><div>Age:<input value=\\"11\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;

exports[`03_usestate should work with recorded events 15`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div>numRendered: 22<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><div>numRendered: 24<div><span>Count: 5</span><button type=\\"button\\">+1</button></div></div><h1>Person</h1><div>numRendered: 42<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bbb\\"></div><div>Age:<input value=\\"111\\"></div></div><div>numRendered: 44<div>First Name:<input value=\\"aaa\\"></div><div>Last Name:<input value=\\"bbb\\"></div><div>Age:<input value=\\"111\\"></div></div></div>
  <script src=\\"/main.js\\"></script>

"
`;
